app/api/courses.js

```diff
@@ -0,0 +1,119 @@
+const { Router } = require('express')
+const course = require('../models/course')
+const router = Router()
+const db = require('../models/index')
+const { logger } = require('../../lib/logger')
+const courseService = require('../services/course_service')
+const enrollmentService = require('../services/enrollment_service')
+const sectionService = require('../services/section_service')
+const { generateUserAuthToken, requireAuthentication } = require('../../lib/auth')
+const e = require('express')
+
+router.use('/', require('./sections'))
+
+//GET request from /courses homepage
+router.get('/', requireAuthentication, async function (req, res) {
+    // TODO: use the authentication token (bearer) to authenticate & find the user
+    const user = await db.User.findByPk(req.payload.sub) // find user by ID, which is stored in sub
+    const teacherCourses = await db.Course.findAll({
+        include: [{
+            model: db.Enrollment,
+            where: { role: 'teacher', userId: user.id }
+        }]
+    })
+    const studentCourses = await db.Course.findAll({
+        include: [
+            {
+                model: db.Section,
+                include: [
+                    {
+                        model: db.Enrollment,
+                        where: { role: 'student', userId: user.id },
+                    }
+                ]
+            }
+        ]
+    })
+
+    res.status(200).send({
+        studentCourses : courseService.extractArrayCourseFields(studentCourses),
+        teacherCourses : courseService.extractArrayCourseFields(teacherCourses)
+    })
+
+})
+
+//User Creates a course
+//Authenticate token, create course & create enrollment for logged in user as teacher in the course
+router.post('/', requireAuthentication, async function (req, res) {
+    const user = await db.User.findByPk(req.payload.sub) // find user by ID, which is stored in sub
+
+    // create course & role as a teacher in enrollment
+    // create course 
+    if (req.body.name) { // name is required, other fields can be left unspecified
```

Comment:
I don't think I perfectly followed this either, but in thinking about it a little while looking this over, it makes sense to create a validate method on any request that has required params/body (even if there's only 1 right now) because it's easier to change in the future if more parameters become required. I'm sure you understand what I mean, but just for thoroughness, in this case we could have a validateCourseCreationSchema method which relies on the courseCreationSchema object. And then if we decided a course also needed a topic, you would just add that to the courseCreationSchema and everything else would work fine and dandy because we'd just be calling the validation method rather than statically defining the required fields in the body like we are here. Nothing to worry about changing but a thought I had moving forward we could try to follow